tapchannel/commitment.go

```diff
@@ -688,34 +687,14 @@ func CreateAllocations(chanState *channeldb.OpenChannel, ourBalance,
 				"sibling: %w", err)
 		}
 
-		// The "incoming" vs. "outgoing" naming is always viewed from
-		// the perspective of the local node and independent of
-		// isOurCommit. So to find out if an HTLC is for the initiator
-		// it only depends on the direction and whether we are the
-		// initiator. The human-readable version of the below switch
-		// statement is: Either we're the initiator and the HTLC is
-		// incoming, or we're not the initiator and the HTLC is
-		// outgoing.
-		var htlcIsForInitiator bool
-		switch {
-		// If the HTLC is incoming and the local node is the channel
-		// initiator, then the HTLC is for the channel initiator.
-		case chanState.IsInitiator && isIncoming:
-			htlcIsForInitiator = true
-
-		// If the HTLC is outgoing and the local node is not the
-		// channel initiator, then the HTLC is for the channel
-		// initiator.
-		case !chanState.IsInitiator && !isIncoming:
-			htlcIsForInitiator = true
-		}
-
-		// We should always just have a single split root on the side
-		// of the initiator. So if the initiator doesn't have any normal
-		// asset output to house the split root, then we'll take the
-		// _first_ HTLC that pays to the initiator.
```

Comment:
In theory we could also enforce an asset reserve requirement as well. I think we'd need some extra TLVs in the `open_channel` and `accept_channel` message to do this properly.

Reply:
True. But I don't see why we'd want to do that? I understand that on the BTC level it's so both parties have something at stake. And since we still have the reserve requirement on the BTC level, that holds true even if we don't also enforce it on the asset level...